ddtrace/tracer.py

```diff
@@ -196,23 +197,28 @@ def global_excepthook(self, tp, value, traceback):
         """The global tracer except hook."""
 
     def get_call_context(self, *args, **kwargs):
```

Comment:
I've done this now in 6880a8d but I'm not 100% about the migration strategy. It's a subtle but significant difference - I think it's important to indicate that the context pertains to a trace (so if there is no trace then there is no context).

We can't really deprecate the API until this point because there are valid use-cases for it. We could suggest using the context_provider directly just so that users stop using it but then the context_provider API is going to change as soon as this PR lands.

Any thoughts or ideas on how to message/deprecate/migrate here @DataDog/apm-python?

Reply:
@P403n1x87 and I discussed this and arrived at this idea: leave in the get_call_context API but return a one-off context when one is not available (different from current_trace_context). This is also a change in behaviour for some use-cases but shouldn't impact most users. The important thing here is that the user's code will not break.

Reply:
is this something we need to continue to be worried about?

it seems like now this API could return a Span correct?

Reply:
oh weird... I thought I had changed the implementation of get_call_context. Indeed it should not return a span and should always return a context.

Reply:
This is a good reminder that we should also leave the tests for get_call_context in place for the time being.

Reply:
Added the tests back